core/server/models/integration.js

```diff
@@ -78,6 +78,28 @@ const Integration = ghostBookshelf.Model.extend({
     webhooks: function webhooks() {
         return this.hasMany('Webhook', 'integration_id');
     }
+}, {
+    permittedOptions(methodName) {
+        let options = ghostBookshelf.Model.permittedOptions.call(this, methodName);
+
+        if (methodName === 'findOne') {
+            options = options.concat(['filter']);
+        }
+
+        return options;
+    },
+
+    findOne(data, unfilteredOptions) {
+        const options = this.filterOptions(unfilteredOptions, 'findOne');
+        data = this.filterData(data);
+        const item = this.forge(data);
+
+        if (options.filter) {
+            item.applyDefaultAndCustomFilters(options);
+        }
+
+        return item.fetch(options);
```

Comment:
Was there a reason you had to avoid calling the base model here?

Reply:
Only that I couldn't see how to 😬

I called `applyDefaultAndCustomFilters()` on a new `this.forge()` instance because it looked like the base model `findOne` calls `this.forge` anyway so the plugin wouldn't be applied if I called the base model? TBH, I have no idea how the plugin system works so it was all an assumption 😄

Reply:
Yeah we don't want to forge the model twice true.

If `filter` is only allowed for integrations, then `filterOptions` will remove `options.filter`.
I think you can just put `item.applyDefaultAndCustomFilters(options);` to base model. That's it.

Reply:
FYI: https://github.com/TryGhost/Ghost/pull/10516

I had to add allowing filter for findOne too. The PR just needs a rebase & tidy up 👍